sns_ik_examples/src/ik_tests.cpp

```diff
@@ -60,20 +60,39 @@ double getDeltaWithLimits(double value, double desired_delta,
   }
 }
 
-bool in_vel_bounds(KDL::JntArray vals, KDL::JntArray vels)
+bool in_vel_bounds(const KDL::JntArray& vals, const KDL::JntArray& vels)
 {
   for(size_t i; i < vels.data.size(); i++){
-      if(vals(i) < -vels(i) || vals(i) > vels(i)){
+      if(vals(i)-1e-6 < -vels(i) || vals(i) > vels(i)+1e-6){
           return false;
       }
-      /*else{
-          std::cout<<vals(i)<<' ';
-      }*/
   }
-  //std::cout<<std::endl;
   return true;
 }
 
+bool in_pos_bounds(const KDL::JntArray& jnt_values, const KDL::JntArray& jnt_lower,
+                   const KDL::JntArray& jnt_upper)
+{
+  for(size_t i; i < jnt_values.data.size(); i++){
+      if(jnt_values(i)-1e-6 < jnt_lower(i) || jnt_values(i) > jnt_upper(i)+1e-6){
```

Comment:
Same thing here. First condition should be (jnt_values(i) < joint_lower(i) - 1e-6)